### DISCUSSION TITLE
Landsat data on Google Cloud

### DISCUSSION THREAD

**Discussion by albaviana** [davidfrantz/force#243] [Q&A] [answered]
Hi all,

I'm working on building annual Landsat BAP composites for Europe and I found that there is no imagery available for 1997 in the Google Cloud storage when executing force-level1-csd. 
Has anyone else found this? It is weird because there are images available for this year in the USGS archive...

Thank you!

Best

Alba.

**Comment by ernstste** [comment, marked as answer]
Hi Alba,

I haven't checked for 1997 data on GCS myself so I'm not sure about the availability.
If possible, you should consider using force-level1-landsat for downloading Landsat data though. GCS only hosts Collection 1 data, C2 is unfortunately not available there. 
force-level1-landsat download capability for Landsat C2 Level1 data was implemented in v3.7.9. You may alternatively use [landsatlinks](https://github.com/ernstste/landsatlinks) in case you're not able to update to the latest version.

Stefan

**Comment by davidfrantz** [comment]
Hey Alba,

I also had the impression that GCS Landsat C1 data has become incomplete... As Stefan said, please think about migrating to these other download possibilities.

Cheers,
David

**Reply by albaviana** [reply]
Hi again,

Thank you @ernstste and @davidfrantz! 

I will definitely migrate to force-level1-landsat but in the meantime I will try landsatlinks.

Best,

Alba.

**Reply by davidfrantz** [reply]
Hey Alba, just to clarify: force-level1-landsat is the same as landsatlinks, just packaged within FORCE.
Cheers,
David

**Reply by albaviana** [reply]
True, thank you! 
I've just started to dive into the info... 

Best

Alba.

**Reply by albaviana** [reply]
Hi David and Stefan,

I finally migrated to force-level1-landsat - landsatlinks to download Landsat data and it is really straightforward, thank you for enabling this! 

Just a short question in case you already know why. It seems that some links are corrupted and therefore the downloading process just stops. I found this in different Landsat scenes and years over Europe. What I am doing is simply going through the generated url text file and deleting them to allow the process to continue... but I wonder if there is a better way to to proceed.

Best,

Alba

**Reply by albaviana** [reply]
Sorry, I forgot to attach the capture!
![Capture](https://user-images.githubusercontent.com/84127033/212911256-80f1282a-f749-4432-a37d-210cca4d01ef.JPG)

what broken links have in common compared to working links is more characters:
BROKEN
https://landsatlook.usgs.gov/gen-bundle?landsat_product_id=LT05_L1TP_202034_19900623_20200916_02_T1&requestSignature=eyJjb250YWN0SWQiOjI2NTg4MDgxLCJkb3dubG9hZElkIjozNTQxODcyOTEsImRhdGVHZW5lcmF0ZWQiOiIyMDIzLTAxLTAzVDA5OjA3OjU5LTA2OjAwIiwic2lnbmF0dXJlIjoiJDUkJHM0eG83ZVVMM1BzWEZmVFwvcnEuZG5jWlpkZGljbjFcLzlnXC9tbmhRSm1pcTIifQ==
WORKING
https://landsatlook.usgs.gov/gen-bundle?landsat_product_id=LT05_L1TP_202034_19900709_20200915_02_T1&requestSignature=eyJjb250YWN0SWQiOjI2NTg4MDgxLCJkb3dubG9hZElkIjozNTQxODcyOTIsImRhdGVHZW5lcmF0ZWQiOiIyMDIzLTAxLTAzVDA5OjA3OjU5LTA2OjAwIiwic2lnbmF0dXJlIjoiJDUkJGdZMDdxWWdcL3FXdVk0NFdsaFVMakdcL29KdzMzVG9qcTE3V2RFaVpKZXpLQSJ9


thank you in advance!

**Comment by ernstste** [comment]
Hi Alba,

there is a rather strict check on download links implemented, which was tailored to how those links were structured. I'm guessing this was changed so I'll adapt the check and possibly relax the check. The download link you posted as broken works fine.

**Reply by ernstste** [reply]
@albaviana this has now been changed in the `davidfrantz/force:latest` docker image. 
You can start the process again to recreate the downloadlinks for the files that you manually deleted. Landsat scenes that are already present in the download folder are skipped automatically.

**Reply by albaviana** [reply]
Hi Stefan,

Thank you for including it so quickly!!
And yes, I actually shared the wrong link...

Best

Alba.
